thorlcr/thorutil/thmem.cpp

```diff
@@ -1471,9 +1471,10 @@ bool CThorSpillableRowArray::_flush(bool force)
     }
 }
 
-bool CThorSpillableRowArray::shrink() // NB: if read active should be protected inside a CThorArrayLockBlock
+bool CThorSpillableRowArray::shrink()
 {
     // NB: Should only be called from writer thread
+    CThorArrayLockBlock block(*this);
```

Comment:
@ghalliday - this is the actual fix. Protecting row array from others.
In particular the race conditional was causing flush() to be active whilst this was without this mutex.